### PR TITLE
Added special case for guessing inner HTML parts for block core/media-text

### PR DESCRIPTION
**Pull request by adrianosferreira**
https://onthegosystems.myjetbrains.com/youtrack/issue/wpmlcore-6074

### CODE REVIEW

**Comment by adrianosferreira** [comment]
@OnTheGoSystems/wpml please review.

File: src/class-wpml-gutenberg-integration.php [hunk 1]
```diff
@@ -257,17 +257,30 @@ private function guess_inner_HTML_parts( $block ) {
...
-				$match_mid_point = 1 + ( strlen( $matches[0] ) - 3 ) / 2;
-				// This is the first ">" char plus half the remaining between the tags
-
-				$parts[0]        .= substr( $matches[0], 0, $match_mid_point );
-				$parts[1]        = substr( $matches[0], $match_mid_point ) . $parts[1];
-			}
+		switch( $block->blockName ) {
+			case 'core/media-text':
+				$html_to_find = '<div class="wp-block-media-text__content">';
+				$pos          = strpos( $inner_HTML, $html_to_find ) + strlen( $html_to_find );
```
Comment (strategio) [inline src/class-wpml-gutenberg-integration.php]: @adrianosferreira, you should use multi-bytes functions `mb_strpos`, `mb_strlen` and `mb_substr`.

We also need to fix this in the `default` logic.

You can add some exotic chars in the tests.

**Review by strategio — CHANGES_REQUESTED** [review]

File: src/class-wpml-gutenberg-integration.php (hunk 1 above)
Comment (adrianosferreira) [reply src/class-wpml-gutenberg-integration.php]: @strategio I don't see the need for that. I can see non-multibyte functions all over the place in the Gutenberg library and 0 occurrences of multibyte ones, so I guess we should change the whole lib. I guess that @BruceOnTheGo should have a reason for that.

Anyway, this code is just a suggestion from Bruce, I believe he had a reason to write using normal string functions.

Comment (strategio) [reply src/class-wpml-gutenberg-integration.php]: @adrianosferreira, I don't see disadvantage using multi-byte functions (some will complain about performance but it's not significant IMO), it's just safer and we already had a lot of bugs related to this.

If changing the functions to multi-byte in the code base does not break the existing tests, I would go for it.

File: src/class-wpml-gutenberg-integration.php [hunk 2]
```diff
@@ -257,17 +257,30 @@ private function guess_inner_HTML_parts( $block ) {
...
-			$parts = explode( $matches[0], $inner_HTML );
-			if ( count( $parts ) === 2 ) {
-				$match_mid_point = 1 + ( strlen( $matches[0] ) - 3 ) / 2;
-				// This is the first ">" char plus half the remaining between the tags
-
-				$parts[0]        .= substr( $matches[0], 0, $match_mid_point );
-				$parts[1]        = substr( $matches[0], $match_mid_point ) . $parts[1];
-			}
+		switch( $block->blockName ) {
+			case 'core/media-text':
```
Comment (strategio) [inline src/class-wpml-gutenberg-integration.php:261]: @adrianosferreira, you wrote a test for this case but if you put a breakpoint on the line below (262), you will see that it never stops.

**Review by strategio — CHANGES_REQUESTED** [review]

File: src/class-wpml-gutenberg-integration.php (hunk 2 above)
Comment (adrianosferreira) [reply src/class-wpml-gutenberg-integration.php:261]: @strategio if the multibyte thing is a blocking issue for you I will hold this fix until @BruceOnTheGo is online to check what he thinks.

I will check the test issue.

Comment (strategio) [reply src/class-wpml-gutenberg-integration.php:261]: I'll leave the multi-bytes suggestion up to you @adrianosferreira (to not delay our release), not blocking.

File: src/class-wpml-gutenberg-integration.php (hunk 1 above)
Comment (brucepearson) [reply src/class-wpml-gutenberg-integration.php]: @strategio I keep forgetting to use the multi-byte functions. @adrianosferreira Please change to use the multi-byte functions. If the existing tests don't break we should be good to go.

Comment (adrianosferreira) [reply src/class-wpml-gutenberg-integration.php]: @strategio adjusted both suggestions, please check.

File: src/class-wpml-gutenberg-integration.php [hunk 3]
```diff
@@ -257,17 +257,30 @@ private function guess_inner_HTML_parts( $block ) {
...
+				$html_to_find = '<div class="wp-block-media-text__content">';
+				$pos          = mb_strpos( $inner_HTML, $html_to_find ) + mb_strlen( $html_to_find );
+				$parts        = array(
+					substr( $inner_HTML, 0, $pos ),
```
Comment (strategio) [inline src/class-wpml-gutenberg-integration.php]: @adrianosferreira, you can use `mb_substr` here and below.

File: src/class-wpml-gutenberg-integration.php [hunk 4]
```diff
@@ -257,17 +257,30 @@ private function guess_inner_HTML_parts( $block ) {
...
+				);
+				break;
+
+			default:
+				preg_match( '#>\s*</#', $inner_HTML, $matches );
+
+				if ( count( $matches ) === 1 ) {
+					$parts = explode( $matches[0], $inner_HTML );
+					if ( count( $parts ) === 2 ) {
+						$match_mid_point = 1 + ( strlen( $matches[0] ) - 3 ) / 2;
```
Comment (strategio) [inline src/class-wpml-gutenberg-integration.php]: And here `mb_strlen`.

**Review by strategio — CHANGES_REQUESTED** [review]

File: src/class-wpml-gutenberg-integration.php (hunk 3 above)
Comment (adrianosferreira) [reply src/class-wpml-gutenberg-integration.php]: done.

File: src/class-wpml-gutenberg-integration.php (hunk 4 above)
Comment (adrianosferreira) [reply src/class-wpml-gutenberg-integration.php]: done.

**Review by strategio — DISMISSED** [review]
Sounds good @adrianosferreira +1

**Review by brucepearson — APPROVED** [review]

**Pull request merged**